From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: oxygencraft <21054297+oxygencraft@users.noreply.github.com>
Date: Sat, 6 Jun 2020 17:23:48 +1000
Subject: [PATCH] Add setters for name and uuid in AsyncPlayerPreLoginEvent

This patch also changes setPlayerProfile to set the name and uuid
of the event to the name and uuid defined in the new player profile.
The name and uuid setters will also set the new name or uuid in the event.

This patch was made because when you set a new player profile, it does
not update the name and uuid in the event. This cause problems when the
event is passed to other listeners that use getName and getUniqueId
which will return the old name and uuid.

diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
index 76c4a1a6e0c0e72b0f3f75d06f179c07f863e143..2bb27554404205badd9543540c13b226b926d9f1 100644
--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
@@ -18,9 +18,9 @@ public class AsyncPlayerPreLoginEvent extends Event {
     private static final HandlerList handlers = new HandlerList();
     private Result result;
     private String message;
-    private final String name;
+    private String name; // Paper - Make not final for setPlayerName and setPlayerProfile
     private final InetAddress ipAddress;
-    private final UUID uniqueId;
+    private UUID uniqueId; // Paper - Make not final for setUniqueId and setPlayerProfile
 
     @Deprecated
     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
@@ -42,11 +42,32 @@ public class AsyncPlayerPreLoginEvent extends Event {
         return profile;
     }
 
-    /*    * Changes the PlayerProfile the player will login as
+    /**
+     * Changes the PlayerProfile the player will login as
      * @param profile The profile to use
      */
     public void setPlayerProfile(@NotNull PlayerProfile profile) {
         this.profile = profile;
+        name = profile.getName();
+        uniqueId = profile.getId();
+    }
+
+    /**
+     * Changes the name of the player they will login as
+     * @param name The name to change it to
+     */
+    public void setPlayerName(@NotNull String name) {
+        this.name = name;
+        profile.setName(name);
+    }
+
+    /**
+     * Changes the unique id the player will login as
+     * @param uniqueId The unique id to use
+     */
+    public void setUniqueId(@NotNull UUID uniqueId) {
+        this.uniqueId = uniqueId;
+        profile.setId(uniqueId);
     }
 
     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile) {
